controllers/snapshot/snapshot_adapter.go

```diff
@@ -334,36 +334,57 @@ TestScenarioLoop:
 // EnsureGlobalCandidateImageUpdated is an operation that ensure the ContainerImage in the Global Candidate List
 // being updated when the Snapshot passed all the integration tests
 func (a *Adapter) EnsureGlobalCandidateImageUpdated() (controller.OperationResult, error) {
-	if (a.component != nil) && gitops.HaveAppStudioTestsSucceeded(a.snapshot) && !gitops.IsSnapshotCreatedByPACPullRequestEvent(a.snapshot) {
-		patch := client.MergeFrom(a.component.DeepCopy())
-		for _, component := range a.snapshot.Spec.Components {
-			if component.Name == a.component.Name {
-				a.component.Spec.ContainerImage = component.ContainerImage
-				err := a.client.Patch(a.context, a.component, patch)
+	if a.component == nil || gitops.IsSnapshotCreatedByPACPullRequestEvent(a.snapshot) {
+		a.logger.Info("The Snapshot wasn't created for a single component push event, not updating the global candidate list.")
+		return controller.ContinueProcessing()
+	}
+	if !gitops.HaveAppStudioTestsSucceeded(a.snapshot) {
+		a.logger.Info("The Snapshot hasn't yet passed all required integration tests, not updating the global candidate list.")
+		return controller.ContinueProcessing()
+	}
+	if gitops.IsSnapshotMarkedAsAddedToGlobalCandidateList(a.snapshot) {
+		a.logger.Info("The Snapshot's component was previously added to the global candidate list, skipping adding it.")
+		return controller.ContinueProcessing()
+	}
+
+	patch := client.MergeFrom(a.component.DeepCopy())
+	for _, component := range a.snapshot.Spec.Components {
+		if component.Name == a.component.Name {
+			a.component.Spec.ContainerImage = component.ContainerImage
```

Comment:
`patch` variable should be within for and if statements

and probably break statement is missing here

Reply:
Ack, my initial idea was to only refactor this to only make the different conditions more distinguishable, but I agree that both of those changes make sense. Updated.